### PR TITLE
Setup GitHub actions

### LINKED ISSUE #41: Set up CI

### LINKED ISSUE #42: Set up testing infrastructure

### PR DESCRIPTION
**Pull request by dinassh**
Changes done in this pull request
- Setup github workflow
- Add end-to-end tests for exercises 0 and 1
- Add a script for generating the github workflow file

### CODE REVIEW

File: exercises/scripts/createE2ETestWorkflow.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
+import { argv } from 'process'
+import { writeFile } from 'fs'
+
+const EXERCISES = ['00-init']
+const WORKFLOW_FILE = '../../.github/workflows/main.yml'
+
+const HEADER = `name: End-to-end tests
```
Comment (novellizator) [inline exercises/scripts/createE2ETestWorkflow.ts]: just to address my OCD and to make it more visually clear - could you add this "name: end-to-end test" on a new line? If you're afraid of an empty initial line, you can use backslash and the end of the line

File: exercises/scripts/createE2ETestWorkflow.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
+import { argv } from 'process'
+import { writeFile } from 'fs'
+
+const EXERCISES = ['00-init']
```
Comment (novellizator) [inline exercises/scripts/createE2ETestWorkflow.ts]: please don't use upper case for some constants, lowercase for others ;) lower-case 'em!

File: exercises/scripts/createE2ETestWorkflow.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    directories.reduce(
+      (accumulator, dir) => accumulator + getCypressJobForDir(dir),
+      ''
+    )
+  )
+}
+
+const cypressWorkflow = getCypressWorkflow(EXERCISES)
+
+if (process.argv.includes('--write')) {
```
Comment (novellizator) [inline exercises/scripts/createE2ETestWorkflow.ts]: just an optional power-up ;-) you could also add a documentation also in the script execution. 
ie  `ts-node createE2ETestWorkflow.ts --help` or -h would tell you the usage

File: exercises/scripts/createE2ETestWorkflow.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+const getCypressWorkflow = (directories: Array<string>): string => {
+  if (directories.length === 0) {
+    return ''
+  }
+
+  return (
+    HEADER +
+    directories.reduce(
```
Comment (novellizator) [inline exercises/scripts/createE2ETestWorkflow.ts]: accumulator x dir --> inconsistent naming (one is short, one is long)

I'd go for "directory"

File: exercises/scripts/createE2ETestWorkflow.ts [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+      (accumulator, dir) => accumulator + getCypressJobForDir(dir),
```
Comment (novellizator) [inline exercises/scripts/createE2ETestWorkflow.ts]: also, i'd probably make an introductory variable for that reduce command, ie

```
const workflowJobs = directories.reduce(...)
return workflowHeader + workflowJobs
```

Comment (novellizator) [reply exercises/scripts/createE2ETestWorkflow.ts]: matter of style (feel free to ignore and discard any comments that I prefix with  "matter of style"): 
usually reduce()s are harder to read than map()s, so many people prefer
`const workflowJobs = directories.map(getCypressJobForDirectory).join()`
also, it's shorter

File: exercises/scripts/createE2ETestWorkflow.ts (hunk 3 above)
Comment (novellizator) [reply exercises/scripts/createE2ETestWorkflow.ts]: i'd also consider renaming the the file to generateE2ETestWorkflow, or generateGithubWorkflowForE2ETests or something like that as "create" is quite general-sounding

File: exercises/00-init/cypress/plugins/index.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
+/// <reference types="cypress" />
```
Comment (novellizator) [inline exercises/00-init/cypress/plugins/index.js]: we seem to have many boilerplate files there. do we need them? (will be need them?)

**Review by novellizator — CHANGES_REQUESTED** [review]
Very well done! Please address the minor issues!

File: exercises/00-init/cypress/support/constants.ts [hunk 7]
```diff
@@ -0,0 +1,9 @@
+export const USER_LIST = {
```
Comment (novellizator) [inline exercises/00-init/cypress/support/constants.ts]: matter of style: i'd probably discard all CAPITALIZED_CONSTANTS as they look ugly to me, but if you prefer them, let's have them. Maybe I'm just too used to swift conventions:)

File: exercises/.gitignore [hunk 8]
```diff
@@ -1 +1,3 @@
-node_modules
\ No newline at end of file
+node_modules
+videos
+
```
Comment (dinassh) [inline exercises/.gitignore:4]: Add `screenshots`  to .gitignore
```suggestion
screenshots

```

File: exercises/scripts/createE2ETestWorkflow.ts (hunk 1 above)
Comment (dinassh) [reply exercises/scripts/createE2ETestWorkflow.ts]: Fixed and renamed, see ff95851 and efff410

File: exercises/scripts/createE2ETestWorkflow.ts (hunk 2 above)
Comment (dinassh) [reply exercises/scripts/createE2ETestWorkflow.ts]: Changed to lowercase, see a2853ed

File: exercises/scripts/createE2ETestWorkflow.ts (hunk 3 above)
Comment (dinassh) [reply exercises/scripts/createE2ETestWorkflow.ts]: Renamed to `generateE2ETestGithubWorkflow.ts`, see 5f6c566

Added help displaying, see 19982517af819a6f2cc938c6a6de2bc416014788

Added path resolving so the script could be run from any directory, see 546c37fb3d6622ba27a5103e89e55220f4b4e0eb

File: exercises/scripts/createE2ETestWorkflow.ts (hunk 4 above)
Comment (dinassh) [reply exercises/scripts/createE2ETestWorkflow.ts]: Rewritten using `map`, see efff410

File: exercises/scripts/createE2ETestWorkflow.ts (hunk 5 above)
Comment (dinassh) [reply exercises/scripts/createE2ETestWorkflow.ts]: Both comments incorporated as proposed, see efff410

I agree, `map` approach is shorter and thanks to that more readable and elegant. On the other hand, the usage of `reduce` states the purpose—iterate through the array and get a single variable of it—a bit more clearly in my opinion. Rewritten for better readability.

File: exercises/00-init/cypress/support/constants.ts (hunk 7 above)
Comment (dinassh) [reply exercises/00-init/cypress/support/constants.ts]: Renamed to `userList`, see a14f4a84503240dc915dafdaa0b22f667b57b09d

I guess I was used to Python conventions :) I have no preference regarding capitalization of constants. I'm happy to adapt my style so it's more readable for others :)

File: exercises/.gitignore (hunk 8 above)
Comment (dinassh) [reply exercises/.gitignore:4]: see 15ca473c628b04b38df88975b08ca3c881f2091d

File: exercises/00-init/cypress/plugins/index.js (hunk 6 above)
Comment (dinassh) [reply exercises/00-init/cypress/plugins/index.js]: Removed, see 9b6e012

**Review by novellizator — APPROVED** [review]

**Review by branoholy — APPROVED** [review]
I'm not familiar with Cypress so I just quickly went through the files.
